contracts/GovernanceFactory.sol

```diff
@@ -97,7 +113,12 @@ contract GovernanceFactory is GovernanceFactoryCreator, UpgradeableContract, ERC
     function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, ERC165) returns (bool) {
         return
             interfaceId == type(GovernanceFactoryCreator).interfaceId ||
-            interfaceId == type(Upgradeable).interfaceId ||
+            interfaceId == type(Versioned).interfaceId ||
             super.supportsInterface(interfaceId);
     }
+
+    /// see UUPSUpgradeable
+    function _authorizeUpgrade(address _caller) internal virtual override(UUPSUpgradeable) onlyOwner {
```

Comment:
trying to understand the flow of this more eg: who the owner will likely be for a community treasury and what the process for upgrading will look like for the community

Reply:
i see the ownership class now, but if you could expand on this^ 🙏

Reply:
There is some documentation for that here [Writing Upgradeable Contracts](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable) and here [UUPSUpgradeable](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/d59306bd06a241083841c2e4a39db08e1f3722cc/contracts/proxy/utils/UUPSUpgradeable.sol#L94)